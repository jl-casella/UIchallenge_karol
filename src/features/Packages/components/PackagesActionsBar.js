import Popup from 'reactjs-popup'
import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

import Button from 'atomics/Button'

const PackagesActionsBarContainer = styled.section`
  display: flex;
  justify-content: space-between;
  width: 100%;
  flex: 0 0 100%;
`

const PackagesActionsBar = ({
  onAddPackage,
  shippingEnabled,
  onShipPackages
}) => (
  <PackagesActionsBarContainer>
    <Button onClick={onAddPackage}>Add Package</Button>
    <Popup
      trigger={
        <Button onClick={onShipPackages} disabled={!shippingEnabled}>
          Ship!
        </Button>
      }
      position="left center"
      on="hover"
      contentStyle={{
        width: 'auto',
        fontWeight: 'normal',
        textAlign: 'center'
      }}
      disabled={shippingEnabled}
    >
      <span>In order to ship packages, all items must be packed</span>
    </Popup>
  </PackagesActionsBarContainer>
)

PackagesActionsBar.propTypes = {
  onAddPackage: PropTypes.func.isRequired,
  shippingEnabled: PropTypes.bool.isRequired,
  onShipPackages: PropTypes.func.isRequired
}

export default PackagesActionsBar
