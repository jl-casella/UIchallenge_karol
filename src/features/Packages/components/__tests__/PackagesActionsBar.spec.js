import { noop } from 'lodash'
import React from 'react'
import { fireEvent, render, waitForElement } from '@testing-library/react'

import PackagesActionsBar from '../PackagesActionsBar'

describe('PackagesActionsBar component test', () => {
  it('Should render correctly', () => {
    const { container } = render(
      <PackagesActionsBar
        onAddPackage={noop}
        shippingEnabled={false}
        onShipPackages={noop}
      />
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('Should call onAdd callback after clicking "Add Package" button', () => {
    const spy = jest.fn()
    const { getByText } = render(
      <PackagesActionsBar
        onAddPackage={spy}
        shippingEnabled={false}
        onShipPackages={noop}
      />
    )

    const addButton = getByText('Add Package')
    expect(spy).toHaveBeenCalledTimes(0)

    fireEvent.click(addButton)
    expect(spy).toHaveBeenCalledTimes(1)
  })

  it('Should call onShip callback after clicking "Ship!" button when shipping enabled', () => {
    const spy = jest.fn()
    const { getByText } = render(
      <PackagesActionsBar
        onAddPackage={noop}
        shippingEnabled={true}
        onShipPackages={spy}
      />
    )

    const shipButton = getByText('Ship!')
    expect(spy).toHaveBeenCalledTimes(0)

    fireEvent.click(shipButton)
    expect(spy).toHaveBeenCalledTimes(1)
  })

  it('Should not call onShip callback after clicking "Ship!" button when shipping disabled', () => {
    const spy = jest.fn()
    const { getByText } = render(
      <PackagesActionsBar
        onAddPackage={noop}
        shippingEnabled={false}
        onShipPackages={spy}
      />
    )

    const shipButton = getByText('Ship!')
    expect(spy).toHaveBeenCalledTimes(0)

    fireEvent.click(shipButton)
    expect(spy).toHaveBeenCalledTimes(0)
  })

  it('Shows information tooltip after hovering over "Ship!" button when shipping disabled', async () => {
    const { getByText } = render(
      <PackagesActionsBar
        onAddPackage={noop}
        shippingEnabled={false}
        onShipPackages={noop}
      />
    )

    const shipButton = getByText('Ship!')
    fireEvent.mouseOver(shipButton)

    await waitForElement(() =>
      getByText('In order to ship packages, all items must be packed')
    )
  })
})
