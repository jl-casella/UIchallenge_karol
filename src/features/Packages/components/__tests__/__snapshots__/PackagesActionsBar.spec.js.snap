// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PackagesActionsBar component test Should render correctly 1`] = `
<section
  class="PackagesActionsBar__PackagesActionsBarContainer-z0x8td-0 iRVhUU"
>
  <button
    class="Button-sc-1bop9qx-0 kLvdUe"
  >
    Add Package
  </button>
  <button
    class="Button-sc-1bop9qx-0 eWZitS"
    disabled=""
  >
    Ship!
  </button>
</section>
`;
